topsort-api-v2.yml

```diff
@@ -0,0 +1,603 @@
+      requestBody:
+        description: 'Use the `/events` endpoint to notify Topsort about significant consumer interactions on the e-commerce site: impressions -- product links become visible to the consumer; clicks -- the consumer clicks on a product link; and purchases -- the consumer buys some products.'
+        content:
+          application/json:
+            schema:
+              oneOf:
+                - $ref: '#/components/schemas/ImpressionEvent'
+                - $ref: '#/components/schemas/ClickEvent'
+                - $ref: '#/components/schemas/PurchaseEvent'
+              discriminator:
+                propertyName: eventType
+                mapping:
+                  Impression: '#/components/schemas/ImpressionEvent'
+                  Click: '#/components/schemas/ClickEvent'
+                  Purchase: '#/components/schemas/PurchaseEvent'
+        required: true
+      responses:
+        200:
+          description:
+            'An object containing the markeplace ID for the event and the Topsort ID for the same event.
+            Logging this can facilitate debugging.
+            The field name for the TopsortID will be "impressionID", "clickId" or "purchaseId"
+            depending on the event in the request.'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/EventResponse'
+        401:
+          $ref: '#/components/responses/UnauthorizedError'
+        400:
+          $ref: '#/components/responses/BadRequest'
+
+components:
+  responses:
+    UnauthorizedError:
+      description: Access token is missing or invalid
+    BadRequest:
+      description: HTTP status codes as registered with IANA.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/TopsortError'
+
+  schemas:
+    AuctionRequest:
+      description: Describes the intent of running a single auction.
+      type: object
+      required:
+        - type
+        - slots
+      discriminator:
+        propertyName: type
+      properties:
+        type:
+          type: string
+          enum:
+            - listings
+            - bannerAds
+          description: Discriminator for the type of auction.
+        slots:
+          type: integer
+          minimum: 1
+          description: Specifies the maximum number of auction winners that should be returned.
+        geoTargeting:
+          $ref: '#/components/schemas/GeoTargeting'
+
+    Category:
+      type: object
+      description: An object describing the category of the bids that will participate in an auction.
+      required:
+        - id
+      properties:
+        id:
+          type: string
+          example: c_yogurt
+
+    SearchQuery:
+      type: string
+      description: The search string provided by a user.
+
+    GeoTargeting:
+      type: object
+      description: An object describing geographical information associated with this auction.
```

Comment:
There must be a clarification indicating how that location is obtained, or how it's configured across their and our system.

Something like, this id is the same that you configured in the Toposort manager.